core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -706,6 +715,71 @@ class SocketServerTest {
     }
   }
 
+  @Test
+  def testSaslReauthenticationFailure(): Unit = {
+    shutdownServerAndMetrics(server) // we will use our own instance because we require custom configs
+    val username = "admin"
+    val password = "admin-secret"
+    val reauthMs = 1500
+    val brokerProps = new Properties
+    brokerProps.setProperty("listeners", "SASL_PLAINTEXT://localhost:0")
+    brokerProps.setProperty("security.inter.broker.protocol", "SASL_PLAINTEXT")
+    brokerProps.setProperty("listener.name.sasl_plaintext.plain.sasl.jaas.config",
+      "org.apache.kafka.common.security.plain.PlainLoginModule required " +
+        "username=\"%s\" password=\"%s\" user_%s=\"%s\";".format(username, password, username, password))
+    brokerProps.setProperty("sasl.mechanism.inter.broker.protocol", "PLAIN")
+    brokerProps.setProperty("listener.name.sasl_plaintext.sasl.enabled.mechanisms", "PLAIN")
+    brokerProps.setProperty("num.network.threads", "1")
+    brokerProps.setProperty("connections.max.reauth.ms", reauthMs.toString)
+    val overrideProps = TestUtils.createBrokerConfig(0, TestUtils.MockZkConnect,
+      saslProperties = Some(brokerProps), enableSaslPlaintext = true)
+    val time = new MockTime()
+    val overrideServer = new TestableSocketServer(KafkaConfig.fromProps(overrideProps), time = time)
+    try {
+      overrideServer.startup()
+      val socket = connect(overrideServer, ListenerName.forSecurityProtocol(SecurityProtocol.SASL_PLAINTEXT))
+
+      val correlationId = -1
```

Comment:
nit: Typically correlationId is positive. Does this have a purpose?

Reply:
Yeah, this is done in multiple places.  For example, twice in `SocketServerTest` (which is where I got the approach from):

https://github.com/apache/kafka/blob/trunk/core/src/test/scala/unit/kafka/network/SocketServerTest.scala#L167
https://github.com/apache/kafka/blob/trunk/core/src/test/scala/unit/kafka/network/SocketServerTest.scala#L687

Twice in `EdgeCaseRequestTest`:
https://github.com/apache/kafka/blob/trunk/core/src/test/scala/unit/kafka/server/EdgeCaseRequestTest.scala#L85
https://github.com/apache/kafka/blob/trunk/core/src/test/scala/unit/kafka/server/EdgeCaseRequestTest.scala#L115

And actually once in non-test code (`AuthorizerUtils`)
https://github.com/apache/kafka/blob/trunk/core/src/main/scala/kafka/security/authorizer/AuthorizerUtils.scala#L91

I do note that the Correlation ID is documented as an integer as opposed to a _positive_ one (https://cwiki.apache.org/confluence/display/KAFKA/A+Guide+To+The+Kafka+Protocol#AGuideToTheKafkaProtocol-Requests).

It feels to me that if this is an issue that needs to be fixed then it might require a separate ticket (especially since there is a non-test occurrence).

Reply:
We're generally ok with micro-cleanups here and there as part of a bug fix, especially since the follow-ups tend not to happen. I see it as akin to the scout's rule of leaving the campground cleaner than how you found it. Anyway, it's not an actual problem; I just thought it was odd.